### PR TITLE
Faster RCNN Generate Proposals

### PR DESCRIPTION
**Pull request by xingyuanbu**

### CODE REVIEW

File: python/paddle/fluid/layers/detection.py [hunk 1]
```diff
@@ -1253,3 +1254,72 @@ def _is_list_or_tuple_(data):
...
     return anchor, var
+
+
+def generate_proposals(scores,
+                       bbox_deltas,
+                       im_info,
+                       anchors,
+                       variances,
+                       pre_nms_topN=6000,
+                       post_nms_topN=1000,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: pre_nms_topN -> pre_nms_top_n
post_nms_topN -> post_nms_top_n

File: python/paddle/fluid/layers/detection.py [hunk 2]
```diff
@@ -1253,3 +1254,72 @@ def _is_list_or_tuple_(data):
...
+                       nms_thresh=0.5,
+                       min_size=0.1,
+                       eta=1.0,
+                       name=None):
+    """
+    ** Generate proposal labels Faster-RCNN **
+	
+	This operation proposes rois according to each box with their probability to be a foreground object and
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: rois -> RoIs

File: python/paddle/fluid/layers/detection.py [hunk 3]
```diff
@@ -1253,3 +1254,72 @@ def _is_list_or_tuple_(data):
...
+ 	2. Calculate box locations as proposals candidates. 
+	3. Clip boxes to image
+	4. Remove predicted boxes with small area. 
+	5. Apply NMS to get final proposals as output.
+	
+      
+	Args:
+		scores(Variable): A 4-D Tensor with shape [N, A, H, W] represents the probability for each box to be an object.
+			N is batch size, A is number of anchors, H and W are height and width of the feature map.
+		bboxDeltas(Variable): A 4-D Tensor with shape [N, 4*A, H, W] represents the differece between predicted box locatoin and anchor location.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: bboxDeltas -> bbox_deltas

File: python/paddle/fluid/layers/detection.py [hunk 4]
```diff
@@ -1253,3 +1254,72 @@ def _is_list_or_tuple_(data):
...
+	5. Apply NMS to get final proposals as output.
+	
+      
+	Args:
+		scores(Variable): A 4-D Tensor with shape [N, A, H, W] represents the probability for each box to be an object.
+			N is batch size, A is number of anchors, H and W are height and width of the feature map.
+		bboxDeltas(Variable): A 4-D Tensor with shape [N, 4*A, H, W] represents the differece between predicted box locatoin and anchor location. 
+		im_info(Variable): A 2-D Tensor with shape [N, 3] represents origin image information for N batch. Info contains height, width and scale
+			between origin image size and the size of feature map.
+		Anchors(Variable):   A 4-D Tensor represents the anchors with a layout of [H, W, num_anchors, 4]. H and W are height and width of the feature map,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: Anchors -> anchors, num_anchors是上面的A吗？ 如果是，写成A吧。

File: python/paddle/fluid/layers/detection.py [hunk 5]
```diff
@@ -1253,3 +1254,72 @@ def _is_list_or_tuple_(data):
...
+              		num_anchors is the box count of each position. Each anchor is in (xmin, ymin, xmax, ymax) format an unnormalized.
+		Variances(Variable): The expanded variances of anchors with a layout of [H, W, num_priors, 4]. Each variance is in (xcenter, ycenter, w, h) format.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: Variances -> variances

File: python/paddle/fluid/layers/detection.py [hunk 6]
```diff
@@ -1253,3 +1254,72 @@ def _is_list_or_tuple_(data):
...
+		pre_nms_topN(float): Number of total bboxes to be kept per image before NMS. 6000 by default.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: pre_nms_topN(float) -> pre_nms_top_n(int)
还少了一个 post_nms_top_n

File: python/paddle/fluid/layers/detection.py (hunk 1 above)
Comment (jerrywgz) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py (hunk 2 above)
Comment (jerrywgz) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py (hunk 3 above)
Comment (jerrywgz) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py (hunk 4 above)
Comment (jerrywgz) [reply python/paddle/fluid/layers/detection.py]: 是A，已改

File: python/paddle/fluid/layers/detection.py (hunk 5 above)
Comment (jerrywgz) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py (hunk 6 above)
Comment (jerrywgz) [reply python/paddle/fluid/layers/detection.py]: Done

**Review by qingqing01 — APPROVED** [review]

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**